objectRecognitionSource/src/test/java/io/vantiq/extsrc/objectRecognition/neuralNet/TestNoProcessorQueries.java

```diff
@@ -0,0 +1,326 @@
+        params.remove("NNfileName");
+        querySource(params);
+        
+        // Check we saved a file in the output directory
+        outputDir = new File(OUTPUT_DIR);
+        assert outputDir.exists();
+        
+        // Check there is only one file, and it's name is equivalent to the last saved file
+        outputDirFiles = outputDir.listFiles();
+        assert outputDirFiles.length == 1;
+        
+        int index = core.lastQueryFilename.lastIndexOf('/') + 1;
+        assert outputDirFiles[0].getName().equals(core.lastQueryFilename.substring(index));
+        
+        // Check that file was saved to Vantiq
+        Thread.sleep(1000);
+        checkUploadToVantiq(core.lastQueryFilename);
+        
+        deleteFileFromVantiq(core.lastQueryFilename);
+        vantiqUploadFiles.add(core.lastQueryFilename);
+        
+        deleteDirectory(OUTPUT_DIR);
+    }
+    
+    // ================================================= Helper functions =================================================
+    
+    public static void setupSource(Map<String,Object> sourceDef) {
+        VantiqResponse insertResponse = vantiq.insert("system.sources", sourceDef);
+        if (insertResponse.isSuccess()) {
+            core = new ObjectRecognitionCore(SOURCE_NAME, testAuthToken, testVantiqServer, MODEL_DIRECTORY);;
+            core.start(CORE_START_TIMEOUT);
+        }
+    }
+    
+    public static Map<String,Object> createSourceDef() {
+        Map<String,Object> sourceDef = new LinkedHashMap<String,Object>();
+        Map<String,Object> sourceConfig = new LinkedHashMap<String,Object>();
+        Map<String,Object> objRecConfig = new LinkedHashMap<String,Object>();
+        Map<String,Object> dataSource = new LinkedHashMap<String,Object>();
+        Map<String,Object> general = new LinkedHashMap<String,Object>();
+        Map<String,Object> neuralNet = new LinkedHashMap<String,Object>();
+        
+        // Setting up dataSource config options
+        dataSource.put("camera", IP_CAMERA_ADDRESS);
+        dataSource.put("type", "network");
+        
+        // Setting up general config options
+        general.put("allowQueries", true);
+        
+        // Setting up neuralNet config options
+        neuralNet.put("type", "none");
+        neuralNet.put("saveImage", "local");
+        neuralNet.put("outputDir", OUTPUT_DIR);
+        
+        // Placing dataSource, general, and neuralNet config options in "objRecConfig"
+        objRecConfig.put("dataSource", dataSource);
+        objRecConfig.put("general", general);
+        objRecConfig.put("neuralNet", neuralNet);
+        
+        // Putting objRecConfig in the source configuration
+        sourceConfig.put("objRecConfig", objRecConfig);
+        
+        // Setting up the source definition
+        sourceDef.put("config", sourceConfig);
+        sourceDef.put("name", SOURCE_NAME);
+        sourceDef.put("type", "ObjectRecognition");
```

Comment:
Should check that this source implementation exists & complain if it's not there.  And probably put a note in the readme that says that we assume it's existence.